include/validator/validator_component.h

```diff
@@ -34,6 +74,51 @@ struct InstanceExprVisitor {
     // and all exports are now specialized to the actual type.
     return {};
   }
+
+private:
+  Context &Ctx;
+
+  std::unordered_map<std::string, ExternDesc> getImports(uint32_t Index) {
+    std::unordered_map<std::string, ExternDesc> ImportMap;
+
+    if (Index >= Ctx.getComponentCount()) {
+      spdlog::error("Unreachable State: Index {} exceeds Component Count {}",
+                    Index, Ctx.getComponentCount());
+      spdlog::error(
+          WasmEdge::ErrInfo::InfoBoundary(Index, 0, Ctx.getComponentCount()));
+
+      return ImportMap;
+    }
+
+    auto &Comp = Ctx.getComponent(Index);
+
+    for (const auto &Sec : Comp.getSections()) {
+      if (std::holds_alternative<ImportSection>(Sec)) {
+        const auto &ImportSec = std::get<ImportSection>(Sec);
+
+        for (const auto &Import : ImportSec.getContent()) {
+          ImportMap[std::string(Import.getName())] = Import.getDesc();
+        }
+      }
+    }
+
+    return ImportMap;
+  }
+
+  bool matchImportAndArgTypes(const ExternDesc &ImportType,
+                              const Sort &ArgSort) {
+    SortCase ArgSortCase = std::get<SortCase>(ArgSort);
+
+    if (std::holds_alternative<TypeBound>(ImportType)) {
+      return ArgSortCase == SortCase::Instance;
+    } else if (std::holds_alternative<ValueType>(ImportType)) {
+      return ArgSortCase == SortCase::Value;
+    } else if (std::holds_alternative<DescTypeIndex>(ImportType)) {
+      return ArgSortCase == SortCase::Type;
+    }
```

Comment:
Better add a link to spec in comment, so even future something changes, maintainers can still compare the implementation and spec.